### DISCUSSION TITLE
Any reason why raspotify would disappear from my connect list after idle?

### DISCUSSION THREAD

**Discussion by mcpattrsn** [dtcooper/raspotify#554] [Q&A] [answered]
I'm a newbie with linux so please be patient with me.

I am using my raspberry pi 4 on raspberry pi os lite for the sole purpose to stream spotify to my DAC. This project works flawlessly and it fits my needs perfectly. So let me get my thanks out of the way for such great work that has been done for this. 

So my problem is that when I am done listening for the night I'll stop play and go to sleep. The next day when I want to listen again my raspberry pi is no longer on my connect list. After a reboot everything shows up and we are back in action. What could be the cause of this?

Again please forgive me if I haven't checked for basic things people more knowledgeable would do. This will be a learning experience for me.

**Comment by JasonLG1979** [comment, marked as answer]
The 1st thing to do is check the logs to see what's going on.

You can do that with:
```
sudo journalctl -u raspotify
```

**Reply by mcpattrsn** [reply]
This is what I get from the latest,,,

Jul 06 19:41:00 raspberrypi systemd[1]: Started Raspotify (Spotify Connect Client).
Jul 06 19:41:00 raspberrypi librespot[437]: [2022-07-07T00:41:00Z WARN  librespot_core::apresolve] Failed to resolve Access Point: error trying to connect: dns error: failed to lookup address information: Tem>
Jul 06 19:41:00 raspberrypi librespot[437]: [2022-07-07T00:41:00Z WARN  librespot_core::apresolve] Using fallback "ap.spotify.com:443"
Jul 06 19:41:00 raspberrypi librespot[437]: [2022-07-07T00:41:00Z ERROR librespot] Connection failed: Cannot create session: failed to lookup address information: Temporary failure in name resolution
Jul 06 19:41:00 raspberrypi systemd[1]: raspotify.service: Main process exited, code=exited, status=1/FAILURE
Jul 06 19:41:00 raspberrypi systemd[1]: raspotify.service: Failed with result 'exit-code'.
Jul 06 19:41:10 raspberrypi systemd[1]: raspotify.service: Scheduled restart job, restart counter is at 1.
Jul 06 19:41:10 raspberrypi systemd[1]: Stopped Raspotify (Spotify Connect Client).
Jul 06 19:41:10 raspberrypi systemd[1]: Started Raspotify (Spotify Connect Client).
Jul 06 20:06:21 raspberrypi librespot[621]: [2022-07-07T01:06:21Z WARN  librespot_playback::player] Player::play called from invalid state

**Reply by JasonLG1979** [reply]
That is an error on Spotify's side. See: https://github.com/librespot-org/librespot/issues/972

Librespot is already falling back to `ap.spotify.com:443` so I'm sorry but I don't think there's much that can be currently done about it?

**Reply by JasonLG1979** [reply]
It's mostly been an issue for European users in the past. I notice that you list your location on Github as Arkansas, are you routing Raspotify though a proxy?

**Reply by mcpattrsn** [reply]
No, however I'm on 5G home internet and it thinks I'm in Denver, or sometimes Nashville when something accesses my location.

**Reply by JasonLG1979** [reply]
I don't know how good your network is? It may just be crappy DNS?

**Reply by JasonLG1979** [reply]
You might maybe try manually setting the DNS server on your Pi so it doesn't use your ISP's DNS servers?

**Reply by mcpattrsn** [reply]
It does have it's issues so I'm now leaning towards that. I checked my router for any automatic updates or restarts, nslookup returns everything just fine, my default dns server is cloudflare, and looking through cron logs it seems like its a carrier issue. The failure happens every day within the same hour, hour and a half. So it could just be my ISP doing something around that time. So I guess I'll just live with it until I get a better option for internet. Oh well, you win some you lose some.

**Reply by JasonLG1979** [reply]
Right, such is life. Sorry about that.

**Reply by mcpattrsn** [reply]
Thank you for all your help!

**Reply by JasonLG1979** [reply]
If it's the same time every day you can always just set a chron job or systemd timer every day at that time to restart the Raspotify service?

**Reply by mcpattrsn** [reply]
There is always that option!

**Reply by JasonLG1979** [reply]
I'm going to mark this as answered. Although the answer is nothing can really be done on our side.
